regnety/dataset/imagenet.py

```diff
@@ -0,0 +1,268 @@
+    ):
+
+        if tfrecs_filepath is None:
+            raise ValueError("List of TFrecords paths cannot be None")
+        self.tfrecs_filepath = tfrecs_filepath
+        self.batch_size = batch_size
+        self.image_size = image_size
+        self.augment_fn = augment_fn
+        self.num_classes = num_classes
+        self.randaugment = randaugment
+        if self.randaugment:
+            self.augmenter = RandAugment(magnitude=5)
+
+    def decode_example(self, example):
+        """Decodes an example to its individual attributes
+
+        Args:
+            example: A TFRecord dataset example.
+
+        Returns:
+            Dict containing attributes from a single example. Follows
+            the same names as TFRECORDS_FORMAT.
+        """
+        image = tf.cast(tf.io.decode_jpeg(example["image"]), tf.float32)
+        height = 512
+        width = 512
+        filename = example["filename"]
+        label = example["label"]
+        synset = example["synset"]
+        return {
+            "image": image,
+            "height": height,
+            "width": width,
+            "filename": filename,
+            "label": label,
+            "synset": synset,
+        }
+
+    def _read_tfrecs(self):
+        """Function for reading and loading TFRecords into a tf.data.Dataset.
+
+        Returns:
+            A tf.data.Dataset
+        """
+        ds = tf.data.TFRecordDataset(self.tfrecs_filepath)
+        ds = ds.map(
+            lambda example: tf.io.parse_example(example, ImageNet.TFRECS_FORMAT)
+        )
+        ds = ds.map(lambda example: self.decode_example(example))
+        return ds
+
+    def _scale_and_center_crop(self, image, h, w, scale_size, final_size):
+        """Resizes image to given scale size and returns a center crop. Aspect
+        ratio is NOT maintained.
+
+        Args:
+            image: tensor of the image
+            h: initial height of image
+            w: initial width of image
+            scale_size: Size of image to scale to
+            final_size: Size of final image
+
+        Returns:
+            Tensor of shape (final_size, final_size, 3)
+        """
+        if w < h and w != scale_size:
+            w, h = tf.cast(scale_size, tf.int64), tf.cast(
+                ((h / w) * scale_size), tf.int64
+            )
+            im = tf.image.resize(image, (w, h))
+        elif h <= w and h != scale_size:
+            w, h = tf.cast(((h / w) * scale_size), tf.int64), tf.cast(
+                scale_size, tf.int64
+            )
+            im = tf.image.resize(image, (w, h))
+        else:
+            im = tf.image.resize(image, (w, h))
+        x = tf.cast((w - final_size) / 2, tf.int64)
+        y = tf.cast((h - final_size) / 2, tf.int64)
+        return im[y : (y + final_size), x : (x + final_size), :]
+
+    def random_sized_crop(self, example, min_area=0.08, max_iter=10):
+        """
+        Randomly chooses an area and aspect ratio and resizes the image to those
```

Comment:
aspect ratio is always square, right?

Reply:
Aspect ratio is a loose term. I mean to say the attribute of the image that it's stretched or squeezed.